### PR TITLE
[Task][Integration] Integrate CRUD in Category

### PR DESCRIPTION
**Pull request by alOletres**
# Issue Link

[Link to the corresponding issue in the github taskboard](https://github.com/orgs/Web-Masters-Freelancing/projects/1/views/1?pane=issue&itemId=55485313)

# Definition of Done

- [x] Users can create category

# Notes
None

# Pre-conditions
run prisma in the server side `npx prisma db push`

# Expected Output
Display of category page and user is able to create category
# Screenshots/Recordings
image

### CODE REVIEW

File: server/prisma/schema.prisma [hunk 1]
```diff
@@ -14,6 +14,11 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
+model Category {
+  id           Int      @id @default(autoincrement())
+  name String    @db.VarChar(100)
+  description String       @db.Text
+}
```
Comment (LodiAniez) [inline server/prisma/schema.prisma:21]: nganong walay relationship sa `Room` na model?

File: server/prisma/schema.prisma [hunk 2]
```diff
@@ -14,6 +14,13 @@ datasource db {
   url      = env("DATABASE_URL")
 }
 
+model Category {
+  id          Int    @id @default(autoincrement())
+  name        String @db.VarChar(100)
+  description String @db.Text
+  Room Room[]
+}
```
Comment (LodiAniez) [inline server/prisma/schema.prisma:22]: pad add ug `isArchived` na column, default value kay false, unsaon nimo pag delete kung wala ni nga column

ayaw idelete kung naay active nga room na gagamit sa category

File: server/src/category/category.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  async create({ name, description }: CreateCategoryDto) {
+    try {
+      await this.prisma.category.create({ data: { name, description } });
+    } catch (err) {
+      catchError(err);
+    }
+  }
+
+  async findAll() {
+    return await this.prisma.category.findMany();
```
Comment (LodiAniez) [inline server/src/category/category.service.ts:19]:
```suggestion
    return await this.prisma.category.findMany({
      where: {
        isArchived: false
      }
    });
```

**Review by LodiAniez — CHANGES_REQUESTED** [review]
borikat, tarungag code

File: server/prisma/schema.prisma (hunk 1 above)
Comment (alOletres) [reply server/prisma/schema.prisma:22]: naa na girl

**Review by LodiAniez — APPROVED** [review]

**Pull request merged**